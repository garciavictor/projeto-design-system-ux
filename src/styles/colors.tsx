export const Colors = {
    primary: {
        900: "#E14624",
        700: "#E4593A",
        500: "#E76B50",
        300: "#EA7E66"
    },
    secondary: {
        900: "#084B59",
        700: "#116F84",
        500: "#1A96B1",
        300: "#24BFE1"
    },
    neutrals: {
        white: {
            100: "#F1F0F0",
            75: "#F5F4F4",
            50: "#F7F7F7",
            25: "#FBFBFB"
        },
        black: {
            100: "#1D1A1A",
            75: "#565353",
            50: "#8D8C8C",
            25: "#C6C5C5"
        }
    },
    success: {
        light: "#4ADE80",
        regular: "#22C55E",
        dark: "#16A34A"
    },
    error: {
        light: "#FF4D4D",
        regular: "#FF0000",
        dark: "#CC0000"
    },
    warning: {
        light: "#FACC15",
        regular: "#EAB308",
        dark: "#CA8A04"
    },
    info: {
        light: "#60A5FA",
        regular: "#3B82F6",
        dark: "#2563EB"
    }
}