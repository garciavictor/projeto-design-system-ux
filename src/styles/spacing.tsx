export const Spacing = {
    s1: 4,
    s2: 8,
    s3: 12,
    s4: 16,
    s5: 20,
    s6: 24,
    s7: 32,
    s8: 40,
    s9: 48,
    s10: 64,
}