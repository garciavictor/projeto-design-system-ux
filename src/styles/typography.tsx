export const Typography = {
    h1: {
        weight: '700',
        size: '48px',
        height: '56px'
    },
    h2: {
        weight: '500',
        size: '36px',
        height: '4px'
    },
    h3: {
        weight: '500',
        size: '28px',
        height: '34px'
    },
    text: {
        weight: '400',
        size: '16px',
        height: '24px'
    },
    small: {
        weight: '200',
        size: '14px',
        height: '20px'
    },
    button: {
        weight: '500',
        size: '16px',
        height: '24px'
    },
    input: {
        text: {
            weight: '500',
            size: '16px',
            height: '24px'
        },
        label: {
            weight: '600',
            size: '16px',
            height: '24px'
        },
        description: {
            weight: '300',
            size: '14px',
            height: '20px'
        }
    }
}