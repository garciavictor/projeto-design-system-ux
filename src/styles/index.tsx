import { BorderRadius } from "./border-radius";
import { Colors } from "./colors";
import { Spacing } from "./spacing";
import { Typography } from "./typography";

export const Theme = {
    typography: Typography,
    colors: Colors,
    spacing: Spacing,
    border_radius: BorderRadius,
}
