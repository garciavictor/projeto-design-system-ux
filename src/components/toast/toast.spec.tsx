import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, act } from '@testing-library/react';
import { Toast, ToastManager, showToast } from '.';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../../styles';

describe('@components/Toast', () => {
    it('matches snapshot for info variant', () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <Toast toastText="Info message" variant="info" />
            </ThemeProvider>
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for success variant', () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <Toast toastText="Success!" variant="success" />
            </ThemeProvider>
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for error variant', () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <Toast toastText="Error occurred!" variant="error" />
            </ThemeProvider>
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for warning variant', () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <Toast toastText="Warning issued!" variant="warning" />
            </ThemeProvider>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('@components/Toast/ToastManager', () => {
    beforeEach(() => {
        vi.useFakeTimers();
    });

    it('renders and displays a toast using showToast()', async () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <ToastManager />
            </ThemeProvider>
        );

        act(() => {
            showToast({ text: 'Test Toast', variant: 'info' });
        });

        expect(container).toMatchSnapshot('Toast rendered after showToast');

        act(() => {
            vi.advanceTimersByTime(3000);
        });

        expect(container).toMatchSnapshot('Toast removed after timeout');
    });
});
