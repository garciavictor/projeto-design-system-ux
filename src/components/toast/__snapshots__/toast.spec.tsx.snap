// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@components/Toast > matches snapshot for error variant 1`] = `
<div
  class="sc-bXdtCk cPaWrr"
  variant="error"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-xmark "
    data-icon="xmark"
    data-prefix="fas"
    focusable="false"
    role="img"
    style="margin-right: 20px;"
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
      fill="currentColor"
    />
  </svg>
  Error occurred!
</div>
`;

exports[`@components/Toast > matches snapshot for info variant 1`] = `
<div
  class="sc-bXdtCk iVmcgw"
  variant="info"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-info "
    data-icon="info"
    data-prefix="fas"
    focusable="false"
    role="img"
    style="margin-right: 20px;"
    viewBox="0 0 192 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M48 80a48 48 0 1 1 96 0A48 48 0 1 1 48 80zM0 224c0-17.7 14.3-32 32-32l64 0c17.7 0 32 14.3 32 32l0 224 32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 512c-17.7 0-32-14.3-32-32s14.3-32 32-32l32 0 0-192-32 0c-17.7 0-32-14.3-32-32z"
      fill="currentColor"
    />
  </svg>
  Info message
</div>
`;

exports[`@components/Toast > matches snapshot for success variant 1`] = `
<div
  class="sc-bXdtCk cWWarp"
  variant="success"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-check "
    data-icon="check"
    data-prefix="fas"
    focusable="false"
    role="img"
    style="margin-right: 20px;"
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
      fill="currentColor"
    />
  </svg>
  Success!
</div>
`;

exports[`@components/Toast > matches snapshot for warning variant 1`] = `
<div
  class="sc-bXdtCk iFnRGl"
  variant="warning"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-exclamation "
    data-icon="exclamation"
    data-prefix="fas"
    focusable="false"
    role="img"
    style="margin-right: 20px;"
    viewBox="0 0 128 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M96 64c0-17.7-14.3-32-32-32S32 46.3 32 64l0 256c0 17.7 14.3 32 32 32s32-14.3 32-32L96 64zM64 480a40 40 0 1 0 0-80 40 40 0 1 0 0 80z"
      fill="currentColor"
    />
  </svg>
  Warning issued!
</div>
`;

exports[`@components/Toast/ToastManager > renders and displays a toast using showToast() > Toast removed after timeout 1`] = `<div />`;

exports[`@components/Toast/ToastManager > renders and displays a toast using showToast() > Toast rendered after showToast 1`] = `<div />`;
